### PR TITLE
Added project name to the PROJECT config file

### LINKED ISSUE #37: Align operator version with the tooling around it
We currently get the operator's version based on git tags, which brings a `v` prefix to the version. Some of the tooling around the operator/kubebuilder expects numeric versions, so, we should adjust the `Makefile` to use the appropriate notations.

Example:

```console
$ make bundle
...
...
FATA[0000] invalid command options: release/v0.2.0-2-gb203820 is not a valid semantic version: Invalid character(s) found in major number "release/v0" 

```

### PR DESCRIPTION
**Pull request by mneverov**
Added project name to the PROJECT config file

projectName field is needed for operator-sdk to deduct the metadata name for the
project. When the bundle command reads input files from the stdin and no --project
argument is provided, and projectName is absent the generated metadata name has
invalid pattern: '.v0.18.0'.
This commit adds the projectName field so metadata.name is correctly generated.

Fixes #37

### CODE REVIEW

**Comment by mneverov** [comment]
```bash
operator-sdk version

operator-sdk version: "v1.3.0-27-gdf8186c4-dirty", commit: "df8186c446c7e8d260bcd42216e59b4a08ce7185", kubernetes version: "v1.19.4", go version: "go1.15.5", GOOS: "linux", GOARCH: "amd64"
```

I believe that error that was mentioned in the original issue was fixed in [PR#4276](https://github.com/operator-framework/operator-sdk/pull/4276).

When I run `make bundle` I get another error:
```bash
make bundle

/home/mneverov/go/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/mneverov/go/bin/operator-sdk generate kustomize manifests -q
/home/mneverov/go/bin/kustomize build config/manifests | /home/mneverov/go/bin/operator-sdk generate bundle -q --overwrite --manifests --version "0.18.0"
ERRO[0000] ClusterServiceVersion validation: [CSVFileNotValid] (.v0.18.0) metadata.name ".v0.18.0" is invalid:
[a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]
ERRO[0000] ClusterServiceVersion validation: [FieldNotFound] required field missing
FATA[0000] Error generating bundle manifests: error generating ClusterServiceVersion: invalid generated ClusterServiceVersion
Makefile:201: recipe for target 'bundle' failed
make: *** [bundle] Error 1
```

ClusterServiceVersion validation: [CSVFileNotValid] (.v0.18.0) **metadata.name** ".v0.18.0" is invalid.

Command `$(OPERATOR_SDK) generate kustomize manifests -q` works because the operator name is [set from the name of the root directory of this project](https://github.com/operator-framework/operator-sdk/blob/df8186c446c7e8d260bcd42216e59b4a08ce7185/internal/cmd/operator-sdk/generate/kustomize/manifests.go#L180) (opentelementry-operator).

Command `$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --overwrite --manifests --version $(VERSION) $(BUNDLE_METADATA_OPTS)` doesn't work because it reads the operator name from the PROJECT file (if it exists), otherwise from the provided `--project` argument.
After I added `projectName` to the `PROJECT` file the `make bundle` was successful:

```bash
make bundle

/home/mneverov/go/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/mneverov/go/bin/operator-sdk generate kustomize manifests -q
/home/mneverov/go/bin/kustomize build config/manifests | /home/mneverov/go/bin/operator-sdk generate bundle -q --overwrite --manifests --version "0.18.0"
/home/mneverov/go/bin/operator-sdk bundle validate ./bundle
INFO[0000] Found annotations file                        bundle-dir=bundle container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=bundle container-tool=docker
INFO[0000] All validation tests have completed successfully
```

I committed all the changes that were done after successful command run. Let me know if I should revert `opentelemetry-operator.clusterserviceversion.yaml` files.
I also run `go mod tidy` and got some changes in `go.sum`.

**Comment by jpkrohling** [comment]
> Let me know if I should revert opentelemetry-operator.clusterserviceversion.yaml files.

Yes, they are generated post-release.

File: config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ metadata:
     operators.operatorframework.io/project_layout: go
     repository: github.com/open-telemetry/opentelemetry-operator
     support: OpenTelemetry Community
-  name: opentelemetry-operator.vX.Y.Z
+  name: opentelemetry-operator.v0.0.0
```
Comment (jpkrohling) [inline config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml:16]: This also looks suspicious.

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -21,11 +21,11 @@ metadata:
     containerImage: quay.io/opentelemetry/opentelemetry-operator
     createdAt: "2020-12-16T13:37:00+00:00"
     description: Provides the OpenTelemetry components, including the Collector
-    operators.operatorframework.io/builder: operator-sdk-v1.2.0
```
Comment (jpkrohling) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:24]: As mentioned in the comments, please revert this file.

File: config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (mneverov) [reply config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml]: the command `$(OPERATOR_SDK) generate kustomize manifests -q` causes it. 
I think it relates to the [PR#4276](https://github.com/operator-framework/operator-sdk/pull/4276) i mentioned.

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (mneverov) [reply bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml]: @jpkrohling TFTR!
reverted the files

File: config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (jpkrohling) [reply config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml]: Got, then it does belong to the PR, right?

Comment (mneverov) [reply config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml:16]: added this change to the PR

**Review by jpkrohling — APPROVED** [review]
Thanks!

**Pull request merged**